AppiumForMac/Server/Controller/AfMSessionController.m

```diff
@@ -574,6 +574,26 @@ -(void) closeWindow
     [self.currentWindow performAction:@"AXCancel"];
 }
 
+-(NSString*)executeShellScript:(NSString*)script
+{
+    FILE *fp;
+    const int BUFFSIZE = 512;
+    char buffer[BUFFSIZE];
+    const char *command = [script UTF8String];
+    NSMutableString *output = [[NSMutableString alloc] initWithString:@""];
+    fp = popen(command, "r");
```

Comment:
Just a question.
What about `launchAndReturnError` instead of `lunch` for NSTask?

Reply:
Not quite following what you mean here.

Reply:
Oh are you saying we use NSTask and the launchAndReturnError method call instead of the deprecated Launch call?

Reply:
Yeah, I saw the `lunch` in https://github.com/appium/appium-for-mac/pull/30/files and wondered if you considered the `launchAndReturnError`. (kindly my curiosity)

Reply:
Just wasn't sure what was going on with the NSTask support if it was going away or will be totally deprecated or what the reason was behind deprecating launch and few other major methods. So I'd thought I'd go with something timeless :P. But if you guys feel like launchAndReturnError is acceptable I'll switch it over.